resources/net.go

```diff
@@ -0,0 +1,726 @@
+func (obj *NetRes) UIDs() []ResUID {
+	x := &NetUID{
+		BaseUID: BaseUID{Name: obj.GetName(), Kind: obj.GetKind()},
+		name:    obj.Name,
+	}
+	return []ResUID{x}
+}
+
+// GroupCmp returns whether two resources can be grouped together or not.
+func (obj *NetRes) GroupCmp(r Res) bool {
+	_, ok := r.(*NetRes)
+	if !ok {
+		return false
+	}
+	return false
+}
+
+// Compare two resources and return if they are equivalent.
+func (obj *NetRes) Compare(r Res) bool {
+	// we can only compare NetRes to others of the same resource kind
+	res, ok := r.(*NetRes)
+	if !ok {
+		return false
+	}
+	if !obj.BaseRes.Compare(res) {
+		return false
+	}
+	if obj.Name != res.Name {
+		return false
+	}
+	if obj.State != res.State {
+		return false
+	}
+	if obj.IPv4 != res.IPv4 {
+		return false
+	}
+	if obj.IPv6 != res.IPv6 {
+		return false
+	}
+	if obj.Gateway != res.Gateway {
+		return false
+	}
+	if obj.Timeout != res.Timeout {
+		return false
+	}
+
+	return true
+}
+
+// UnmarshalYAML is the custom unmarshal handler for this struct.
+// It is primarily useful for setting the defaults.
+func (obj *NetRes) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawRes NetRes // indirection to avoid infinite recursion
+
+	def := obj.Default()     // get the default
+	res, ok := def.(*NetRes) // put in the right format
+	if !ok {
+		return fmt.Errorf("could not convert to NetRes")
+	}
+	raw := rawRes(*res) // convert; the defaults go here
+
+	if err := unmarshal(&raw); err != nil {
+		return err
+	}
+
+	*obj = NetRes(raw) // restore from indirection with type conversion!
+	return nil
+}
+
+// Close cleans up some resources so we can exit cleanly.
+func (obj *NetRes) Close() error {
+	// close the dbus connection
+	obj.sdbus.Close()
+	// close the message bus
+	obj.gdbus.Close()
+
+	return obj.BaseRes.Close()
+}
+
+// unitFileContents builds the unit file contents from the input
+func unitFileContents(name, ipv4, ipv6, gateway string) []byte {
+	// build the unit file contents
+	u := "[Match]\n"
+	u += "Name=" + name + "\n"
+	u += "[Network]\n"
+	if ipv4 == "" && ipv6 == "" {
+		u += "DHCP=yes\n"
+	}
+	if ipv4 != "" {
+		u += "Address=" + ipv4 + "\n"
+	}
+	if ipv6 != "" {
+		u += "Address=" + ipv6 + "\n"
+	}
+	if gateway != "" {
+		u += "Gateway=" + gateway + "\n"
+	}
+	return []byte(u)
+}
+
+// getAddrsFromIface returns the provided interface's IPv4 and IPv6 addresses.
+// If an address isn't found, an empty string is returned in its place.
+func getAddrsFromIface(name string) (ipv4, ipv6 string, _ error) {
+	// find the interface
+	iface, err := net.InterfaceByName(name)
+	if err != nil {
+		return "", "", errwrap.Wrapf(err, "error finding interface: %s", name)
+	}
+
+	// check the interface's ip addresses
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return "", "", errwrap.Wrapf(err, "error getting addrs")
+	}
+	for _, addr := range addrs {
+		// ipv4
+		if strings.Contains(addr.String(), ".") {
```

Comment:
Cute-- Is this the way to check for ipv4 vs. v6?

In any case, as discussed if IP is an array, this just becomes easier, and we don't need to distinguish :)
If need be someone could write an `isIPv4` func for the language.

Reply:
My first pass used this. https://golang.org/pkg/net/#IP.To4 but ACK on the array (slice?)

Reply:
Oh! Cool--- Why not use that?